### PR TITLE
Create release notes for Wazuh 3.12.0

### LINKED ISSUE #2316: Create release notes for Wazuh 3.12.0
Hello team! This issue aims to create the release notes for Wazuh 3.12.0.

Regards,

David

### PR DESCRIPTION
**Pull request by davidfmiranda**
Hello team!

This PR closes #2316 

## Description

I have added release notes for Wazuh 3.12.0 version: 

release_notes

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 

Regards,

David

### CODE REVIEW

File: source/release-notes/release_3_12_0.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
+.. Copyright (C) 2020 Wazuh, Inc.
+
+.. _release_3_12_0:
+
+3.12.0 Release notes
+====================
+
+This section lists the changes in version 3.11.4. More details about these changes are provided in each component changelog:
```
Comment (alberpilot) [inline source/release-notes/release_3_12_0.rst]: Wrong version

File: source/release-notes/release_3_12_0.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+- `wazuh/wazuh-api <https://github.com/wazuh/wazuh-api/blob/3.12/CHANGELOG.md>`_
+- `wazuh/wazuh-ruleset <https://github.com/wazuh/wazuh-ruleset/blob/3.12/CHANGELOG.md>`_
+
+
+Wazuh core
+----------
+
+**File integrity monitoring**
+
+- Add synchronization capabilities for FIM.
```
Comment (alberpilot) [inline source/release-notes/release_3_12_0.rst]: Added instead Add

File: source/release-notes/release_3_12_0.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+- Add SQL database for the FIM module. Its storage can be switched between disk and memory.
+- Add FIM module unit testing for Unix source code.
+- Add FIM module unit testing for Windows source code.
+- Move the FIM logic engine to the agent.
+
+**Logcollector**
+
+- Avoid reopening the current socket when Logcollector fails to send an event.
```
Comment (alberpilot) [inline source/release-notes/release_3_12_0.rst]: Past tense?

Comment (davidfmiranda) [reply source/release-notes/release_3_12_0.rst]: Done

**Pull request merged**